lib/Pill/index.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Icon from '../Icon';
+
+const Pill = props => {
+  return (
+    <span className={`pill ${props.className}`}>
+      {props.children}
+      {props.onClearClick ? (
+        <button
+          type="button"
+          onClick={props.onClearClick}
+          className="button--clear button--icon-only-padding"
```

Comment:
Any reason why you're using a `button` with the button classes on it? Seems like you wanna use the `Button` proxy component?

Reply:
The `Button` proxy came with a lot of extra classes that didn't work well with `button--icon-only`. It turns out whenever we use the class `button--icon-only` or `button--icon-only-padding`, it is on an `a` and not a `Button`